service/operator/service.go

```diff
@@ -387,23 +402,25 @@ func (s *Service) GetDeleteState(obj, currentState, desiredState interface{}) (i
 		{
 			newData := map[string]string{}
 			for k, v := range deleteState.ConfigMap.Data {
-				if inConfigMapData(dState.ConfigMapData, k, v) {
+				if !inConfigMapData(dState.ConfigMapData, k, v) {
```

Comment:
The result from `KeyFromTPO(obj)` will be put into some structure. This structure will be returned by `GetDeleteState`. This structure will then be passed to `ProcessDeleteState` and the rest of your code is done in there.